### ISSUE TITLE
Serde feature enabled by default (through `std` feature)

### ISSUE THREAD

**Issue by phil-opp** [00imvj00/mqttrs#30]
The `serde` feature is active by default because the `std` feature depends on it:

https://github.com/00imvj00/mqttrs/blob/17cc76e1ada477feeae72820345ba1e2144376c8/Cargo.toml#L17-L22

This makes it impossible to build the library with only the `std` feature, but not the `serde` feature. Also, the Readme states that you have to add the `derive` feature in order to use serde, which is not required.

Ideally, we would want to enable the `serde/std` feature only if both the `std` and `derive` features are active, but I don't know if that's possible.

(This is not a real problem for me at the moment, I just though that this looks like a bug, so I decided to report it. Thanks for creating this library, it looks very promising from a first look!)

**Comment by vincentdephily** [comment]
Good catch, thanks. I pushed a quick fix to https://github.com/00imvj00/mqttrs/tree/fix_feature_hierarchy but didn't test it beyond looking at `cargo build` and `cargo tree` with various features enabled.

I'm not sure how to properly unittest feature-dependant code. Maybe using an examples folder ? How do I test for the absence of a feature ?

**Comment by phil-opp** [comment]
Great, thanks! One possible approach for testing could be to create an example that fails to compile if the serde feature is enabled (e.g. using [`compile_error!`](https://doc.rust-lang.org/stable/std/macro.compile_error.html)). Then you can try to compile it with the `std` feature to ensure that the serde feature is not enabled implicitly.
